File: frontend_mobile/lib/data/models/recipe/recipe_summary_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pairing_planet2_frontend/domain/entities/recipe/recipe_summary.dart';

part 'recipe_summary_dto.g.dart';

@JsonSerializable(explicitToJson: true)
class RecipeSummaryDto {
  final String publicId;
  final String foodName;
  final String foodMasterPublicId;
  final String title;
  final String? description;
  final String? culinaryLocale;
  final String? creatorPublicId; // Creator's publicId for profile navigation
  final String? userName;
  final String? thumbnail;
  final int? variantCount;
  final int? logCount; // Activity count from backend (nullable for backward compat)
  final String? parentPublicId;
  final String? rootPublicId;
  final String? rootTitle; // Root recipe title for variants
  final int? servings;
  final String? cookingTimeRange;
  final List<String>? hashtags; // Hashtag names (first 3)

  RecipeSummaryDto({
    required this.publicId,
    required this.foodName,
    required this.foodMasterPublicId,
    required this.title,
    this.description,
    this.culinaryLocale,
    this.creatorPublicId,
    required this.userName,
    this.thumbnail,
    this.variantCount,
    this.logCount,
    this.parentPublicId,
    this.rootPublicId,
    this.rootTitle,
    this.servings,
    this.cookingTimeRange,
    this.hashtags,
  });

  factory RecipeSummaryDto.fromJson(Map<String, dynamic> json) =>
      _$RecipeSummaryDtoFromJson(json);

  Map<String, dynamic> toJson() => _$RecipeSummaryDtoToJson(this);

  RecipeSummary toEntity() => RecipeSummary(
    publicId: publicId,
    foodName: foodName,
    foodMasterPublicId: foodMasterPublicId,
    title: title,
    description: description ?? "",
    culinaryLocale: culinaryLocale ?? "",
    thumbnailUrl: thumbnail,
    creatorPublicId: creatorPublicId,
    userName: userName ?? "익명",
    variantCount: variantCount ?? 0,
    logCount: logCount ?? 0,
    parentPublicId: parentPublicId,
    rootPublicId: rootPublicId,
    rootTitle: rootTitle,
    servings: servings ?? 2,
    cookingTimeRange: cookingTimeRange ?? 'MIN_30_TO_60',
    hashtags: hashtags,
  );
}

File: backend/src/main/java/com/cookstemma/cookstemma/dto/recipe/RecipeSummaryDto.java
package com.cookstemma.cookstemma.dto.recipe;

import java.util.List;
import java.util.UUID;

/**
 * Recipe summary DTO with pre-localized title/description.
 * The title and description fields contain values for the requested locale,
 * resolved server-side from the translations map.
 */
public record RecipeSummaryDto(
        UUID publicId,
        String foodName,
        UUID foodMasterPublicId,
        String title,           // Localized title (resolved from titleTranslations)
        String description,     // Localized description (resolved from descriptionTranslations)
        String cookingStyle,
        UUID creatorPublicId,   // Creator's publicId for profile navigation
        String userName,
        String thumbnail,
        Integer variantCount,
        Integer logCount,       // Activity count: number of cooking logs
        UUID parentPublicId,
        UUID rootPublicId,
        String rootTitle,       // Root recipe title for variants (lineage display)
        Integer servings,       // Number of servings
        String cookingTimeRange, // Cooking time range enum
        List<String> hashtags   // Hashtag names (first 3)
) {}